### DISCUSSION TITLE
Support for Wi-Fi suggestion API on Android Q (10)

### DISCUSSION THREAD

**Discussion by Minuit-Z** [ThanosFisherman/WifiUtils#127] [Ideas]
As [wifi-suggect](https://developer.android.google.cn/guide/topics/connectivity/wifi-suggest) said: 
> Devices running Android 10 (API level 29) and higher allow your app to add network credentials for a device to auto-connect to a Wi-Fi access point. You can supply suggestions for which network to connect to using WifiNetworkSuggestion

we can use `WifiNetworkSuggestion` api to connect wifi on Android Q+. 
but I think it`s not friendly for developers .  we can just provider a suggestion to tell the system: I wanna  connect this ssid. Then the system would  receive it " at a right time " .

I tried this way many times , but this method works by accident.  when it works , the system would send a notification for users to choose weather we use " this suggestion ". Uses may ignore such a suggestion or choose to reject .

**Comment by ThanosFisherman** [comment]
WiFiUtils is currently using the IoT API for connecting to wifi networks which limits the internet connectivity only to the app that requested it.

I haven't really studied the suggestion API yet but it seems like a good alternative although it might have its limitations as well. I think it's worth a try migrating to suggestion API once I get the hang of it.

**Comment by aboozaid** [comment]
It seems WifiNetworkSuggestion works very well if your device is not connected to a network regardless of it being saved  network or not

**Reply by yichen8095** [reply]
You save my time. Many kinds of devices startScan always return false, so it failed to connect to a network all the time. 
Wi-Fi suggestion API does work!
